gui/system/forms.py

```diff
@@ -2705,6 +2717,19 @@ class Meta:
         model = models.Certificate
 
 
+def is_valid_privatekey(certificate, privatekey):
+    pkey_obj = crypto.load_privatekey(crypto.FILETYPE_PEM, privatekey)
+    cert_obj = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
+    ctx = SSL.Context(SSL.TLSv1_METHOD)
```

Comment:
Yes please add it for CAs as well.

When you say that "the code breaks for RSA keys" what exactly do you mean? How you generate these RSA keys? Are they bare RSA 1024 (or whatever length) RSA pub/priv key pairs? Those will not work with this verification anyways. Please elaborate this statement further.